common/changes/@rushstack/heft-config-file/user-danade-JestPlugin_2021-06-08-01-29.json

```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/heft-config-file",
+      "comment": "Add \"preresolve\" property to transform paths before resolution",
```

Comment:
The reason I went this route is that it allows for much finer-grained control over how we want to resolve the path before going into the node module resolution. Only providing token replacement doesn't allow for much flexibility in the future. Token replacement isn't something that we're going to do any more efficiently or with less code than any other dev would, so might as well give more options rather than box it into a specific design.

Reply:
That's a reasonable point. It may end up being a really complicated feature to implement that doesn't end up working for a lot of cases.